tools/common/StaticRegistrar.cs

```diff
@@ -496,8 +496,22 @@ public string ToObjCType (TypeDefinition type)
 			if (IsNativeObject (type))
 				return "id";
 
-			if (IsDelegate (type))
-				return "id";
+			if (IsDelegate (type)) {
+				if (!delegateToBlockType)
```

Comment:
Did you try to always return the block signature instead of just `id` (i.e. skip the `delegateToBlockType` check and go straight to the code below? If it didn't work, do you remember why?

Reply:
I tried always returning the block signature, but then I got build errors. The reason is this:

```objc
- (void)doSomething:(void (^)(BOOL))paramName; // This is valid syntax.
void doSomethingElse(void (^)(BOOL) paramName); // This is a syntax error.
void doSomethingElse(void (^paramName)(BOOL)); // This is valid syntax.
```

Without the delegateToBlockType check, we get the second output in the above example, and a broken build as a result. I will try threading the parameter name into the method so we get the third output, but I don't know if it will work. Thanks!

Reply:
Unfortunately, this won't really work. If I were to move the parameter name into the type, I would then need to ensure that it is no emitted after the type as well [here](https://github.com/xamarin/xamarin-macios/blob/cd5917650e40cab7ed70cd75db3efe8220c60ba4/tools/common/StaticRegistrar.cs#L3988-L3989). I could do this, but then I'd have to implement an out-parameter from the ToObjCParameterType method to tell the caller that the parameter was already emitted, which IMHO isn't worth the increased complexity. Marshalling block types as `id` everywhere other than in Objective-C method declarations works just fine.